File: app/src/main/java/ch/berta/fabio/tipee/dialogs/CountryNotDetectedDialogFragment.java
package ch.berta.fabio.tipee.dialogs;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

import ch.berta.fabio.tipee.R;

/**
 * A {@link android.app.DialogFragment} subclass that displays a dialog informing the user that
 * the country could not be automatically detected.
 *
 * @author Fabio Berta
 */
public class CountryNotDetectedDialogFragment extends DialogFragment {
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
        dialogBuilder.setTitle(R.string.country_not_detected_dialog_title)
                .setMessage(R.string.country_not_detected_dialog_message)
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dismiss();
                    }
                });
        return dialogBuilder.create();
    }
}

File: app/src/main/java/ch/berta/fabio/tipee/features/tip/dialogs/CountryNotDetectedDialogFragment.kt
package ch.berta.fabio.tipee.features.tip.dialogs

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.support.v4.app.FragmentManager
import android.support.v7.app.AlertDialog
import ch.berta.fabio.tipee.R

/**
 * A [android.app.DialogFragment] subclass that displays a dialog informing the user that
 * the selectedCountry could not be automatically detected.

 * @author Fabio Berta
 */
class CountryNotDetectedDialogFragment : DialogFragment() {

    companion object {
        val tag: String = CountryNotDetectedDialogFragment::class.java.canonicalName

        fun display(fm: FragmentManager) {
            val dialog = CountryNotDetectedDialogFragment()
            dialog.show(fm, tag)
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialogBuilder = AlertDialog.Builder(activity)
        dialogBuilder.setTitle(R.string.country_not_detected_dialog_title)
                .setMessage(R.string.country_not_detected_dialog_message)
                .setPositiveButton(R.string.ok) { _, _ -> dismiss() }
        return dialogBuilder.create()
    }
}
